framework/src/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -116,17 +136,17 @@
      * @param path
      *            The current tree path.
      */
-    public static FlowExpressions. /*@Nullable*/ Receiver parse(String s,
+    public static FlowExpressions. /*@Nullable*/ Receiver parseAllowingItself(String s,
```

Comment:
Why do you write 'FlowExpressions.Receiver' instead of just 'Receiver'? The latter must be in scope, as seen on line 148. The latter would be much easier to read and avoid the awkward type annotation position.